### PR TITLE
docs: Standardize optional type in docstrings

### LINKED ISSUE #374: Standardize Optional Type in docstrings
Currently a parameter or return type of `some_type | None` is described in the docstring as either `some_type | None` or `Optional[some_type]`. We should set our standard for this case and change all other cases to this new standard.

Additionally there are cases where the parameter allows a None type but the docstring type does not show this. We should add this information to all those cases.

## Examples
`Optional[type]` in the `Imputer`
```py
 def fit(self, table: Table, column_names: list[str] | None) -> Imputer:
        """
        [...]
        Parameters
        ----------
        [...]
        column_names : Optional[list[str]]
            The list of columns from the table used to fit the transformer. If `None`, all columns are used.
        [...]
        """
```
`type | None` in `Table`:
```py
    def __init__(self, data: Mapping[str, Sequence[Any]] | None = None) -> None:
        """
        Create a table from a mapping of column names to their values.

        Parameters
        ----------
        data : Mapping[str, Sequence[Any]] | None
            The data. If None, an empty table is created.
        [...]
        """
```
Example of a method without this information in `Table.py`:
```py
def slice_rows(
        self,
        start: int | None = None,
        end: int | None = None,
        step: int = 1,
    ) -> Table:
        """
        Slice a part of the table into a new table.

        This table is not modified.

        Parameters
        ----------
        start : int
            The first index of the range to be copied into a new table, None by default.
        end : int
            The last index of the range to be copied into a new table, None by default.
        [...]
        """
```

### PR DESCRIPTION
**Pull request by jxnior01**
Closes #374 

### Summary of Changes

* Selected `optional type` and standardized it in docstrings

### CODE REVIEW

**Comment by Marsmaennchen221** [comment]
Didn´t we say we want the docs to be the same as the implementation so we don´t want `Optional[...]` and we do want `... | None` in the implementation as well as in the docstrings?

File: docs/development/guidelines.md [hunk 1]
```diff
@@ -265,6 +265,32 @@ The docstrings **should** use the [numpydoc](https://numpydoc.readthedocs.io/en/
...
+    ```
+
+!!! failure "**DON'T** (library code):"
+
+    ```py
+    def __init__(self, data: Mapping[str, Any] | None = None) -> None:
+        """
+        data : Optional[Mapping[str, Any]]
+        """
+
```
Comment (Marsmaennchen221) [inline docs/development/guidelines.md:284]:
```suggestion
    ```

```

File: docs/development/guidelines.md [hunk 2]
```diff
@@ -265,6 +265,32 @@ The docstrings **should** use the [numpydoc](https://numpydoc.readthedocs.io/en/
...
+!!! failure "**DON'T** (library code):"
+
+    ```py
+    def __init__(self, data: Mapping[str, Any] | None = None) -> None:
+        """
+        data : Mapping[str, Any] | None
+        """
+
```
Comment (Marsmaennchen221) [inline docs/development/guidelines.md:292]:
```suggestion
    ```
```

**Review by Marsmaennchen221 — CHANGES_REQUESTED** [review]

**Review by Marsmaennchen221 — CHANGES_REQUESTED** [review]

**Review by alex-senger — CHANGES_REQUESTED** [review]
There is still an "optional docstring" in `InvertibleTableTransformer.fit()` in `_table_transformer.py`

**Review by alex-senger — APPROVED** [review]
[LGTM](https://lgtm.party/p/1N7T2VRO/)

**Review by Marsmaennchen221 — CHANGES_REQUESTED** [review]
Looks good so far.

You missed the docstrings in method `Table.slice_rows`

In `regression.AddaBoost` please change the docstring for `AdaBoost.learner` to return a `Regressor`

**Comment by jxnior01** [comment]
> > Looks good so far.
> > You missed the docstrings in method `Table.slice_rows`
> > In `regression.AddaBoost` please change the docstring for `AdaBoost.learner` to return a `Regressor`
> 
> Adressed these issues. Also merged current `main` into this.

Thanks. I'm rounding up with issue #150. I planned resolving all reviews when i'm done.

**Review by Marsmaennchen221 — APPROVED** [review]
[LGTM](https://lgtm.party/p/2MISV3O/)

**Pull request merged**